rotkehlchen/chain/evm/decoding/rainbow/decoder.py

```diff
@@ -308,74 +90,154 @@ def _process_swap(self,
-            if (unwrap_event := self._find_event_by_log(all_logs, WETH_WITHDRAW_TOPIC)) is None:
-                log.error(
-                    f'Failed to decode fee from rainbow swap involving'
-                    f'token to {self.evm_inquirer.native_token} for transaction '
-                    f'{transaction} due to missing unwrap log.',
-                )
-                return decoded_events
+        # update the note with the actual value. This is the amount used by the rainbow
+        # router to swap and there could be an actual lower amount swapped since solvers
+        # also charge some amount.
+        out_event.notes = f'Swap {out_event.amount} {out_event.asset.resolve_to_asset_with_symbol().symbol} in Rainbow'  # noqa: E501
+        in_event.notes = f'Receive {in_event.amount} {in_event.asset.resolve_to_asset_with_symbol().symbol} as the result of a swap in Rainbow'  # noqa: E501
+        return decoded_events
 
-            self._maybe_decode_fee_from_swap_token_to_native(
+    def _find_fees(
+            self,
+            out_event: 'EvmEvent',
+            in_event: 'EvmEvent',
+            transaction: 'EvmTransaction',
+            decoded_events: list['EvmEvent'],
+            all_logs: list['EvmTxReceiptLog'],
+    ) -> None:
+        """Check transfers to find the fee taken from rainbow. They can bee ETH fees in which
+        case we check the internal transactions or token fees where we check in the log events
+        """
+        swapped_amount = out_event.amount
+        # if we are dealing with eth swaps check the internal transfers of eth from/to
+        # the rainbow router
+        if self.evm_inquirer.native_token in {out_event.asset, in_event.asset}:
+            for internal_tx in DBEvmTx(self.evm_inquirer.database).get_evm_internal_transactions(
+                parent_tx_hash=transaction.tx_hash,
+                blockchain=self.evm_inquirer.blockchain,
+            ):
+                # the fee is the amount diff with the amount that the proxy sends to
+                # the swap solver
+                if ((
+                    internal_tx.from_address == self.router_address and
+                    internal_tx.to_address != out_event.location_label
+                ) or (
+                    internal_tx.from_address != out_event.location_label and
+                    internal_tx.to_address == self.router_address
+                )):
+                    swapped_amount = from_wei(FVal(internal_tx.value))
+                    break
+        # When eth is swapped the fee is taken from the sent ETH. When ETH is received the
+        # free is taken from the received amount.
+        if (
+            out_event.asset == self.evm_inquirer.native_token and
+            (fee_amount := out_event.amount - swapped_amount) > 0
+        ):
+            out_event.amount -= fee_amount
+            decoded_events = self._create_and_append_fee_event(
```

Comment:
```suggestion
            self._create_and_append_fee_event(
```
Same for the other places this function is called.